src/internal/operators/startWith.ts

```diff
@@ -7,6 +7,7 @@ import { isScheduler } from '../util/isScheduler';
 import { MonoTypeOperatorFunction, SchedulerLike } from '../types';
 
 /* tslint:disable:max-line-length */
+export function startWith<T>(scheduler?: SchedulerLike): MonoTypeOperatorFunction<T>;
```

Comment:
I think we're going to need to differentiate between what's being passed into the `startWith` and what the source observable is to really get this right. :thinking: 

```ts
export function startWith<T, R>(...args: R[], scheduler?: SchedulerLike): OperatorFunction<T, T|R>;
```

Reply:
@benlesh Yeah, you are right. This is very similar to the changes that were made regarding the default parameters in `first` and `last` - in https://github.com/ReactiveX/rxjs/pull/3952/commits/04f5f0b3499e0c3fdf4088d5fbe8f90d534c6480 - that effect a union type if a non-`T` default is specified.

Reply:
AFAICT this won't be work because it's only possible to define a rest parameter as the last argument in a function. 
I think it should be doable if we solve it the same way as in the commit @cartant mentioned.

Reply:
@timdeschryver You'd need to use a union type as the rest parameter type. Have a look at [`concat`](https://github.com/ReactiveX/rxjs/blob/6.2.2/src/internal/observable/concat.ts#L15).

Reply:
I just noticed that there is [already a rest parameters signature](https://github.com/ReactiveX/rxjs/blob/6.2.2/src/internal/operators/startWith.ts#L16). I think the `T` in the signatures should be replaced with an `R` - where `R` defaults to `T`, as in the referenced commit. And the return type should be `OperatorFunction<T, T | R>` which will collapse to `OperatorFunction<T, T>` when `R` is `T`.

Reply:
Yea, I agree that's what I have right now.
The only difference is that I have a type for each argument, I'll commit and push so you can take a look and give your opinions on it.

Reply:
Yeah, I guess if `R` doesn't have to be `T`, having one type parameter per parameter makes sense.